### PR TITLE
Splitting functionality and code quality check tests

### PR DESCRIPTION
**Pull request by heemanshusuri**

### CODE REVIEW

File: .azure-pipelines/functionality-test.yml [hunk 1]
```diff
@@ -19,13 +19,6 @@ variables:
   value: $(Build.BuildNumber)
```
Comment (ninghu) [inline .azure-pipelines/functionality-test.yml]: Shall we rename to `function-tests.yml`?

File: .azure-pipelines/code-quality.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      targetType: 'inline'
+      script: |
+        pip install --user azureml-sdk
+        R -e 'dir.create(Sys.getenv("R_LIBS_USER"), recursive = TRUE);
+              .libPaths(Sys.getenv("R_LIBS_USER"));
+              # build and install R sdk
+              devtools::install_deps(upgrade = FALSE);
+              package_location <- devtools::build();
+              install.packages(package_location, repos = NULL)'
+    displayName: 'Build and Install SDK'
```
Comment (ninghu) [inline .azure-pipelines/code-quality.yml]: I think we can remove this step to save some time as R CMD check would make sure SDK is build-able.

File: .azure-pipelines/functionality-test.yml (hunk 1 above)
Comment (ninghu) [inline .azure-pipelines/functionality-test.yml]: For variables start with `TEST_`, shouldn't they change to read from environment variable so we can control behavior in build?

Comment (heemanshusuri) [reply .azure-pipelines/functionality-test.yml]: how about sdk-core.yml?

Comment (ninghu) [reply .azure-pipelines/functionality-test.yml]: Hmm.. can't tell it's for tests from the its name.

File: .azure-pipelines/function-tests.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      targetType: 'inline'
+      script: |
+        pip install --user azureml-sdk
+        R -e 'dir.create(Sys.getenv("R_LIBS_USER"), recursive = TRUE);
+              .libPaths(Sys.getenv("R_LIBS_USER"));
+              # build and install R sdk
+              devtools::install_deps(upgrade = FALSE);
+              package_location <- devtools::build();
+              install.packages(package_location, repos = NULL)'
+    displayName: 'Build and Install SDK'
```
Comment (ninghu) [inline .azure-pipelines/function-tests.yml]: in order to run testthat tests, do we need to build and install R sdk? just curious

File: .azure-pipelines/update-docs.yml [hunk 4]
```diff
@@ -31,8 +31,6 @@ steps:
         R -e 'dir.create(Sys.getenv("R_LIBS_USER"), recursive = TRUE);
               .libPaths(Sys.getenv("R_LIBS_USER"));
               devtools::install_deps(upgrade = FALSE);
-              unlink("man", recursive=TRUE, force=TRUE);
```
Comment (ninghu) [inline .azure-pipelines/update-docs.yml]: I already have a PR for this. You can either approve my PR or manually merge the changes to this PR.
https://github.com/Azure/azureml-sdk-for-r/pull/182/files

Comment (heemanshusuri) [reply .azure-pipelines/update-docs.yml]: sounds good

File: .azure-pipelines/function-tests.yml (hunk 3 above)
Comment (heemanshusuri) [reply .azure-pipelines/function-tests.yml]: I guess tests won't work if you don't have azuremlsdk package pre-built.

Comment (ninghu) [reply .azure-pipelines/function-tests.yml]: Got it.

**Pull request merged**